### PR TITLE
Test normal and precompiled gap

### PR DESCRIPTION
**Pull request by sebasguts**

### CODE REVIEW

**Comment by fingolfin** [comment]
Please rebase so that CI runs again and we get proper coverage.

**Comment by sebasguts** [comment]
Done.

File: .travis.yml [hunk 1]
```diff
@@ -4,10 +4,18 @@ julia:
...
- global:
-   - GAPROOT=$TRAVIS_BUILD_DIR/gap
-   - MAKEFLAGS="-j4"
-   - JULIA_TRACK_COVERAGE=yes
+  - GAP=$HOME/.julia/gap.sh
+    GAP_INSTALL_PACKAGES=debug
+    MAKEFLAGS="-j4"
+    JULIA_TRACK_COVERAGE=yes
+    CFLAGS+=--coverage
+    LDFLAGS+=--coverage
```
Comment (fingolfin) [inline .travis.yml]: Huh, are you sure += is allowed here?!? I kinda doubt it

Comment (fingolfin) [reply .travis.yml]: Looking at the build logs, it definitely doesn't work.

Also, why not keep the common env vars in an "env: global:" ? That's what it's for, no?

Comment (sebasguts) [reply .travis.yml]: I will try. It did not work for me the last time I tried. However, it should also work without the `+`.

Comment (sebasguts) [reply .travis.yml]: Lets see how it works now.

File: .travis.yml [hunk 2]
```diff
@@ -28,11 +34,11 @@ before_script:
   - printenv | sort
   - git show --pretty=fuller -s
   - mkdir -p coverage
-  - etc/travis_install_gap.sh
+  - if [[ -v GAPROOT ]]; then etc/travis_install_gap.sh; fi
```
Comment (fingolfin) [inline .travis.yml]: More idiomatic would be
```suggestion
  - [[ -v GAPROOT ]] && etc/travis_install_gap.sh
```

File: .travis.yml [hunk 3]
```diff
@@ -28,11 +34,11 @@ before_script:
...
     ## Add the package so we can use it later
   - julia -e 'import Pkg ; Pkg.add(Pkg.PackageSpec(url=pwd()))'
   - julia -e 'import Pkg ; Pkg.add("Nemo")'
-  - etc/travis_build.sh
+  - if [[ -v GAPROOT ]]; then etc/travis_build.sh; fi
```
Comment (fingolfin) [inline .travis.yml]:
```suggestion
  - [[ -v GAPROOT ]] && etc/travis_build.sh
```

**Comment by sebasguts** [comment]
Argh, now I merged it, but we still have incorrect displayed data, right?

**Comment by fingolfin** [comment]
Well, we'll just need to make another PR with coverage fixes :-). Besides missing some .c file, this is also missing all files from GAP.jl itself, i.e., from the /src/ dir.

**Comment by sebasguts** [comment]
Oha, yeah, I also have some suspicion where that comes from. I will make another PR.

**Pull request merged**